PolymorphismWithTemplates/main.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <iomanip>
+
+
+//---------------------------------------------------------------------------------------------------------------------
+// Data definitions
+//---------------------------------------------------------------------------------------------------------------------
+
+struct Rectangle
+{
+   int width;
+   int height;
+};
+
+struct Book
+{
+   std::string author;
+   std::string title;
+};
+
+struct Secret
```

Comment:
Create `struct`s that represent the data to log. If a single type needs to be formatted two different ways, create two `struct`s to represent the two formats.